### PR TITLE
Allow static home page template override.

### LINKED ISSUE #1745: Have way of having an `/about` page used as front page
Cobrands would like the ability to have a static page as the front page, with the reporting page shown elsewhere. They don't want to have to override the index.html template and then recreate that somewhere else.

### PR DESCRIPTION
**Pull request by dracos**
If you have a templates/web/COBRAND/about/homepage.html file, then that page will be used as the front page of your site, and the normal front page will instead be available at `/report`. Fixes #1745 

- [x] Tests

### CODE REVIEW

File: perllib/FixMyStreet/App/Controller/Report.pm [hunk 1]
```diff
@@ -28,14 +28,14 @@ Redirect to homepage unless C<id> parameter in query, in which case redirect to
...
 
-    my $id = $c->get_param('id');
-
-    my $uri =
-        $id
-      ? $c->uri_for( '/report', $id )
-      : $c->uri_for('/');
-
-    $c->res->redirect($uri);
+    if (my $id = $c->get_param('id')) {
```
Comment (davea) [inline perllib/FixMyStreet/App/Controller/Report.pm]: How old is this functionality? Is it still in use anywhere or could we ruthlessly chop it out?

File: templates/web/base/main_nav_items.html [hunk 2]
```diff
@@ -1,4 +1,4 @@
-[%~ INCLUDE navitem uri='/' label=loc('Report a problem') attrs='class="report-a-problem-btn"' ~%]
+[%~ INCLUDE navitem uri=(homepage_uri OR '/') label=loc('Report a problem') attrs='class="report-a-problem-btn"' ~%]
```
Comment (davea) [inline templates/web/base/main_nav_items.html]: `homepage_uri` isn't always set, so e.g. on `/auth` the 'Report a problem' link is to `/`, not `/report`. This seems to be a bug, unless I'm misunderstanding something perhaps.

File: perllib/FixMyStreet/App/Controller/Report.pm (hunk 1 above)
Comment (dracos) [inline perllib/FixMyStreet/App/Controller/Report.pm]: Looks like .com forces `/report` to redirect to `/reports` anyway, for typos, so we've never used it ourselves.
I don't see it being used anywhere, `/around` does it for e.g. front page form, so yeah, could strip it I assume.

File: templates/web/base/main_nav_items.html (hunk 2 above)
Comment (dracos) [inline templates/web/base/main_nav_items.html]: No, quite correct! Hmm, guess it'll have to look it up on each request, then.

**Pull request merged**